src/mgconfitem.h

```diff
@@ -122,12 +122,31 @@ class MLITESHARED_EXPORT MGConfItem : public QObject
     */
     QStringList listDirs() const;
 
+    /*! Return a map of the keys inside this directory. The
+        returned strings are absolute key names like
+        "/myapp/settings".
+
+        A directory is a key that has children.  The same key might
+        also have a value, but that is confusing and best avoided.
+    */
+
+    QStringList listKeys() const;
+
+    /*! Return a map of the keys with it's values inside this directory.
+
+        A directory is a key that has children.  The same key might
+        also have a value, but that is confusing and best avoided.
+    */
+
+    QVariantMap listValues() const;
```

Comment:
Can't `listValues()` be done (with possibly slightly worse performance if all values are required, but possibly better performance if only some values are required) on the client side by something like:

```
MGConfItem foo(...);
QVariantMap values;
foreach (const QString &key, foo.listKeys()) {
    values[key] = MGConfItem(key).value();
}
```

This way, getting only values of interest is easily possible:

```
MGConfItem foo(...);
QVariantMap values;
foreach (const QString &key, foo.listKeys()) {
    if (key.contains("something_interesting")) {
        values[key] = MGConfItem(key).value();
    }
}
```

If so, we might want to not provide a `listValues()` function in the API until we are sure this is needed (basically if there are enough use cases where performance would benefit compared to the loop-over-keys-and-fetch-value construct).